core/common/src/main/scala/org/typelevel/otel4s/TextMapPropagator.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+trait TextMapPropagator[F[_]] extends TextMapInjector[F] with TextMapExtractor
+
+trait TextMapInjector[F[_]] {
```

Comment:
On the extractor side, we are to return a new immutable Context, so it's pure.  No such constraint is given for injection, and the Java one returns Unit.